core/store/src/trie/mod.rs

```diff
@@ -757,6 +760,97 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    ///
+    ///  Whenever a proof is not found, an empty vector is returned.
+    ///
+    /// # Errors
+    ///
+    /// If the root is not found on the trie, a `StorageError` will be returned
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<Vec<TrieProofItem>, StorageError> {
+        let mut key = NibbleSlice::new(key);
+
+        let mut hash = *root;
+        if hash == Trie::empty_root() {
+            return Ok(vec![]);
+        }
+
+        let mut levels: Vec<TrieProofItem> = vec![];
+        loop {
+            let bytes = self.storage.retrieve_raw_bytes(&hash)?;
+            let node = RawTrieNodeWithSize::decode(&bytes).map_err(|_| {
+                StorageError::StorageInconsistentState("RawTrieNode decode failed".to_string())
```

Comment:
- given that we are encoding a string here.  Should we capture the returned error code here to aid in debugging if this fires in production?
- Should we also encode some other bits of local context here to aid debugging in production?

Reply:
FYI, I have https://github.com/near/nearcore/pull/7233.  Once it lands this code should use function it introduces.